### PR TITLE
comm with js: pipe operators

### LINKED ISSUE #19: Document pipe first

### PR DESCRIPTION
**Pull request by jchavarri**
Fixes #19.

Based on #32 (`comm-with-js/lang-concepts`).

Adds a subsection "Pipe operators" to the "Language concepts" section.

I decided to explain OCaml pipe operator, and then explain the limitations that led to the introduction of pipe first operator. I tried to avoid references to existing content besides the blog post I wrote, as all the issues and PRs that I could find are super confusing.

In general, it is quite hard to explain this in a way that makes sense, but hopefully it helps to have a reference that other places in the docs can point to. The PR already updates the links from the "Chaining" section.

### CODE REVIEW

**Comment by jchavarri** [comment]
By the way, I could not find _any_ documentation of `|>` in the OCaml manual 😮

**Comment by anmonteiro** [comment]
> By the way, I could not find _any_ documentation of `|>` in the OCaml manual 😮

It might only be documented in the stdlib https://v2.ocaml.org/api/Stdlib.html#1_Compositionoperators

File: docs/communicate-with-javascript.md [hunk 1]
```diff
@@ -176,6 +176,162 @@ For a comprehensive understanding of abstract types and their usefulness, refer
...
+String.cat [ 1; 2; 3 ]`, the type mismatch is detected when the list is type
+checked, after `String.cat` has been processed.
+
+With the goal of addressing this kind of limitations, Melange introduces the
+pipe first operator `|.`.
+
+#### Pipe first
+
+To overcome the constraints mentioned above and enable the generation of more
+performant JavaScript code, Melange introduces the pipe first operator `|.`.
```
Comment (anmonteiro) [inline docs/communicate-with-javascript.md:279]: what's an example of a code snippet that is more performant with pipe first?

**Review by anmonteiro — APPROVED** [review]
looks good but I think the "more performant" claim is misleading, if not plain wrong.

File: docs/communicate-with-javascript.md (hunk 1 above)
Comment (jchavarri) [reply docs/communicate-with-javascript.md:279]: https://github.com/rescript-lang/rescript-compiler/issues/1671

Comment (jchavarri) [reply docs/communicate-with-javascript.md:281]: I see. Updated docs in https://github.com/melange-re/melange-re.github.io/pull/50.

**Pull request merged**